contracts/plugins/assets/RTokenAsset.sol

```diff
@@ -128,8 +139,46 @@ contract RTokenAsset is IAsset, VersionedAsset {
 
     // solhint-enable no-empty-blocks
 
+    function forceUpdatePrice() external {
+        _updateCachedPrice();
+    }
+
+    function latestPrice() external returns (uint192 rTokenPrice, uint256 updatedAt) {
+        // Situations that require an update, from most common to least common.
+        if (
+            cachedOracleData.cachedAtTime + ORACLE_TIMEOUT <= block.timestamp || // Cache Timeout
+            cachedOracleData.cachedAtNonce != basketHandler.nonce() || // Basket nonce was updated
+            cachedOracleData.cachedTradesNonce != backingManager.tradesNonce() || // New trades were started
+            cachedOracleData.cachedTradesOpen != backingManager.tradesOpen() // ..or settled
+        ) {
+            _updateCachedPrice();
+        }
+
+        return (cachedOracleData.cachedPrice, cachedOracleData.cachedAtTime);
+    }
+
     // ==== Private ====
 
+    // Update Oracle Data
+    function _updateCachedPrice() internal {
+        if (cachedOracleData.cachedAtTime == block.timestamp) {
+            // The price was updated in the same block.
+            return;
+        }
+
+        (uint192 low, uint192 high) = price();
```

Comment:
Noting out loud that when we bring the lotPrice changes over to this repo this _should_ stay `price()` and _should not_ change to `lotPrice()`.